### PR TITLE
Add msw code gen and tidy up handlers

### LINKED ISSUE #541: Service worker handlers cleanup and enable workign in tandem with the backend
At the moment the service worker is off in development (I think 🙄 ?)

i.e. uses 

```
if (process.env.NODE_ENV === 'development') {
  const {
    setupMockWorker,
    // eslint-disable-next-line @typescript-eslint/no-var-requires
  } = require('@openmsupply-client/mock-server/src/worker/client');
  const worker = setupMockWorker();

  worker.start();
}
```

- We should change this to instead, have a toggle to disable the worker completely (This gives the option of using the apollo server)
- By default, disable the worker in development for some queries but not others - i.e. querying for requisitions, stocktakes - this is so we can run the backend on the demo site and then have the worker support the pages that are unsupported by the backend.

Tidy up:
- A general tidy up of the `handlers.ts` file - splitting into functional areas similar to how the mock server is set up. Essentially, the `handlers` function just needs to replace the `schema/Invoice.ts` files etc.
- If we can somehow also support using specific handlers in tests that would be amazing - i.e. if I wanted to write a jest test for the `ItemSearchInput`, it would be great if I could provide a list of items in my test to a handler, so the worker intercepts the request within the jest test and I can then test whether those items were rendered, rather than the current implementation where it is a random set of data generated

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #541 

Sigh.. this isn't going to be my default from now on, I swear - it's just easier this way 😢 

- I've added the msw code gen package - it's useful, but not amaaazing as I had hoped 😁  - the types can be annoying
- I shifted the handlers into their own directories
- I added some basic dev/prod handling 
- An example of how to use handlers in tests, for now

### CODE REVIEW

File: packages/mock-server/src/worker/handlers/requisition/index.ts [hunk 1]
```diff
@@ -0,0 +1,297 @@
+import {
```
[the remark this answers is not in the document]
Comment (joshxg) [reply packages/mock-server/src/worker/handlers/requisition/index.ts:1]: maybe I should have done more files. Though I don't really expect much more queries/mutations so should be ok

File: packages/requisitions/src/types.ts [hunk 2]
```diff
@@ -40,11 +40,11 @@ export interface SupplierRequisition extends Requisition {
...
   isDeleted: boolean;
   otherParty: Name;
-  lines: SupplierRequisitionLine[];
+  lines: CustomerRequisitionLine[];
   update: (key: string, value: string) => void;
   updateOtherParty: (value: Name) => void;
-  upsertLine?: (line: SupplierRequisitionLine) => void;
-  deleteLine?: (line: SupplierRequisitionLine) => void;
+  upsertLine?: (line: CustomerRequisitionLine) => void;
+  deleteLine?: (line: CustomerRequisitionLine) => void;
```
Comment (joshxg) [inline packages/requisitions/src/types.ts:47]: Oh. Is this an offroad fix for something

**Comment by mark-prins** [comment]
I've tried running as production ( a "start-prod" script.. `webpack-cli serve --env production` )
and am having issues.. 😢 
- The `stockCounts` and `invoiceCounts` handlers have gone and are required to be picked up by msw. Ideally the invoice would be in the `invoice\index.ts` handler.. but.. since the other parts are handled by the real api.. what about an 'experimental' handler file? or 'extras'? or..?
- Error: Unknown type "StocktakeListParameters" ( looking into this now.. )
- Error: Unknown type "RequisitionListParameters" ( ditto )

**Comment by joshxg** [comment]
Ah crap.. sorry! Of course the dashboard queries! 😠

**Comment by joshxg** [comment]
Have pushed the experimental file - not getting the unfound parameters, will take a deeper look. 

(Can also bring the supported/unsupported arrays down to the feature-level if we want to support/not support more specific within a feature)

**Comment by joshxg** [comment]
Oh right. Yes I am getting it when I build for prod 🙄  process.env.NODE_ENV is not to set to production when I run your script.

It must be set somehow on the demo server during a build 🤔

**Review by mark-prins — APPROVED** [review]
👍  That is too good!! just what I was hoping for
not sure what is up with running through webpack-cli with prod env. but who cares? 
have built a prod version and am running that on demo 😁 

have added a change to `webpack.config` - to copy the `mockServiceWorker.js` to `dist`. I'd rather not say how long it took me to realise that 🙄

**Comment by joshxg** [comment]
Ah.. of course, copying over the worker! oops..

Also I hope that diff is just being set to the correct value and I haven't doubled the size inadvertently 😁

**Comment by mark-prins** [comment]
🤔  The bundle diff size worried me.. but it was the first run, so I assumed it was wrong.
However.. if I build 0bf76483 then that give me `1,723,351 bytes` in my `dist` folder and if I build the latest I get `3,688,966 bytes` Maybe it is right.. will check

**Comment by mark-prins** [comment]
Oh. 🤦  of course. The big difference is that `faker` is now in the production build, and that's `1.41 MB`
There's a bit more but it all seems legit

**Comment by joshxg** [comment]
Oh, of course! Wow faker is massive!!

**Pull request merged**